starlette/testclient.py

```diff
@@ -91,11 +97,16 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
         await instance(receive, send)
 
 
+class _AsyncBackend(TypedDict):
```

Comment:
I'm not really convinced this `TypedDict` is really worth it over just `Dict[str, Any]` 🤔. Definitely debatable (and I guess a matter of taste), but what class of bug is this really helping with?

Reply:
mypy processes this for the **async_backend call

Reply:
Enables autocomplete functionalities (depending on env environment), and type checking tools are able to detect invalid keys e.g. mypy.

Reply:
> mypy processes this for the anyio.start_blocking_portal(**self.async_backend)

I didn't know it could do that. That is kinda neat.

> Enables autocomplete functionalities (depending on plugins..), and type checking tools are able to detect invalid keys e.g. mypy.

Sure, but..it's an internal interface with 2 keys...

Anyway, this isn't blocking my approval, just my 2c.